packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -143,15 +167,23 @@ export class ConnectionService {
       protocol: HandshakeProtocol.Connections,
       role: ConnectionRole.Inviter,
       multiUseInvitation: false,
-      routing,
+      did,
+      mediatorId,
       autoAcceptConnection: outOfBandRecord.autoAcceptConnection,
       tags: {
         invitationKey: outOfBandRecord.getTags().recipientKey,
       },
     })
 
-    connectionRecord.theirDid = message.connection.did
-    connectionRecord.theirDidDoc = message.connection.didDoc
+    // TODO Could we use authentication key from did doc instead of did comm services?
+    const theirVerkeys = message.connection.didDoc?.didCommServices[0].recipientKeys ?? []
```

Comment:
This is a tricky one. RecipientKeys is where we should _send_ messages to, but doesn't necesarily mean we will receive messages using these keys. So for completeness we should probably use all authentication keys (as legacy ed25519) and all keyagreement keys?

Reply:
Hmm, I see. To me it seems the reason we're storing the recipient keys here is to later find did doc based on sender keys. We can always find other keys via did doc services if needed somewhere else. I agree we should use keys from more didcomm services but I would rather store just authentication keys.